crates/sui-framework/packages/sui-framework/sources/accumulator.move

```diff
@@ -69,6 +71,142 @@ fun root_remove_accumulator<K, V: store>(accumulator_root: &mut AccumulatorRoot,
 
 use fun root_remove_accumulator as AccumulatorRoot.remove_accumulator;
 
+/// === Accumulator metadata ===
+///
+/// Accumulator metadata is organized as follows:
+/// - Each address that holds at least one type of accumulator has an owner object attached
+///   to the accumulator root.
+/// - For each type of accumulator held by that address, there is an AccumulatorMetadata object
+///   attached to the owner object.
+/// - When the value of an accumulator drops to zero, the metadata object is removed.
+/// - If the owner object has no more accumulator metadata objects attached to it, it is removed
+///   as well.
+
+public struct OwnerKey has copy, drop, store {
+    owner: address,
+}
+
+/// An owner object, to which all AccumulatorMetadata objects for the owner are
+/// attached.
+public struct Owner has store {
+    /// The individual balances owned by the owner.
+    balances: bag::Bag,
```

Comment:
Was there a reason to use `Bag` instead of `table`? I think `Bag` is for storing data of different types, but if all key/values are of the same type, we should use table

Reply:
shouldn't we use a bag in case we ever need to add a new type of child field here? Or will we be able to change it later? I'm not clear on the evolvability rules, it seems conceptually possible since Bag and Table have the same exact layout

Reply:
Oh, actually, this is not possible anyway. The key used here is `MetadataKey<T>`, so each field does in fact have a unique type.